optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/stream/ConstraintStreamScoreDirectorFactory.java

```diff
@@ -62,13 +65,21 @@ public ConstraintStreamScoreDirectorFactory(SolutionDescriptor<Solution_> soluti
     // ************************************************************************
 
     @Override
-    public ConstraintStreamScoreDirector<Solution_> buildScoreDirector(
-            boolean lookUpEnabled, boolean constraintMatchEnabledPreference) {
+    public ConstraintStreamScoreDirector<Solution_> buildScoreDirector(boolean lookUpEnabled,
+            boolean constraintMatchEnabledPreference) {
         return new ConstraintStreamScoreDirector<>(this, lookUpEnabled, constraintMatchEnabledPreference);
     }
 
     public ConstraintSession<Solution_> newConstraintStreamingSession(boolean constraintMatchEnabled, Solution_ workingSolution) {
         return constraintSessionFactory.buildSession(constraintMatchEnabled, workingSolution);
     }
 
+    // ************************************************************************
+    // Getters/setters
+    // ************************************************************************
+
+    public Constraint[] getConstraints() {
+        return Arrays.copyOf(constraints, constraints.length);
```

Comment:
ConstraintStreamScoreDirectorFactory.getConstraints is non-public API.
We avoid defensive copies in non-public API, unless it's a probably-one-day-public-API, to avoid performance surprises and to reduce the code size.
This method will never be part of a public API I'd argue.